File: android/app/src/main/java/com/pablomartin/encriptador/LocalizationRu.kt
package com.pablomartin.encriptador

internal object LocalizationRu {
    val mapa: Map<String, String> = mapOf(
        // ===== common =====
        "common.mostrar" to "Показать",
        "common.ocultar" to "Скрыть",
        "common.contrasena" to "ПАРОЛЬ",
        "common.repetirContrasena" to "ПОВТОРИТЕ ПАРОЛЬ",
        "common.footer.credito" to "Разработано: %s",
        "common.estado.passwordRequerida" to "Введите пароль.",
        "common.estado.passwordsNoCoinciden" to "Пароли не совпадают.",
        "common.estado.archivoEncriptado" to "Файл успешно зашифрован.",
        "common.estado.archivoDesencriptado" to "Файл успешно расшифрован.",
        "common.estado.verificandoPassword" to "Проверка пароля...",
        "common.estado.error" to "Ошибка: %s",

        // ===== strength =====
        "strength.muyDebil" to "Очень слабый",
        "strength.debil" to "Слабый",
        "strength.media" to "Средний",
        "strength.fuerte" to "Сильный",
        "strength.muyFuerte" to "Очень сильный",

        // ===== crypto =====
        "crypto.archivoDanado" to "Неверный пароль или повреждённый файл.",
        "crypto.tipoEsCarpeta" to "Этот .enc содержит папку.",
        "crypto.tipoEsArchivo" to "Этот .enc — отдельный файл.",

        // ===== pantalla (MainActivity) =====
        "pantalla.tagline" to "Шифрование файлов AES-256-GCM",
        "pantalla.archivoOCarpeta" to "ФАЙЛ ИЛИ ПАПКА",
        "pantalla.btnArchivos" to "📄 Файл(ы)",
        "pantalla.btnCarpeta" to "📁 Папка",
        "pantalla.nadaSeleccionado" to "Ничего не выбрано",
        "pantalla.variosSeleccionados" to "Выбрано файлов: %d",
        "pantalla.carpetaSeleccionada" to "Папка: %s",
        "pantalla.conservarOriginal" to "Сохранить исходный файл (не удалять по завершении)",
        "pantalla.conservarEnc" to "Сохранить файл .enc (не удалять по завершении)",
        "pantalla.btnEncriptar" to "🔒 Зашифровать",
        "pantalla.btnDesencriptar" to "🔑 Расшифровать",
        "pantalla.btnElegirCarpetaDestino" to "📂 Выбрать папку назначения и извлечь",
        "pantalla.progresoArchivo" to "Файл %1\$d из %2\$d",
        "pantalla.elegirDondeGuardar" to "Выберите, куда сохранить .enc...",
        "pantalla.sinSeleccion" to "Выберите файл(ы) или папку.",
        "pantalla.variosArchivosEnc" to "%d файлов.enc",
        "pantalla.variosEncriptados" to "Файлов зашифровано в один пакет: %d.",
        "pantalla.noSePudoAbrirCarpeta" to "Не удалось открыть папку.",
        "pantalla.carpetaEncriptadaUnArchivo" to "Папка зашифрована в один файл.",
        "pantalla.soloUnArchivoEnc" to "Выберите один файл .enc.",
        "pantalla.carpetaConArchivos" to "Папка содержит файлов: %d. Выберите, куда извлечь.",
        "pantalla.extrayendo" to "Извлечение...",
        "pantalla.noSePudoAbrirDestino" to "Не удалось открыть место назначения.",
        "pantalla.carpetaRestaurada" to "Папка восстановлена: %d файл(ов).",
        "pantalla.errorAlExtraer" to "Ошибка при извлечении: %s",
        "pantalla.noSePudoCrearCarpeta" to "Не удалось создать папку «%s».",
        "pantalla.noSePudoCrearArchivo" to "Не удалось создать файл «%s».",
        "pantalla.nombreFallbackArchivo" to "файл",
        "pantalla.nombreFallbackCarpeta" to "папка",
    )
}

File: linux/Encriptador/Services/Localization.Ru.cs
namespace Encriptador.Services;

public static partial class Loc
{
    private static readonly Dictionary<string, string> Ru = new()
    {
        // ===== common =====
        ["common.mostrar"] = "Показать",
        ["common.ocultar"] = "Скрыть",
        ["common.encriptar"] = "Зашифровать",
        ["common.desencriptar"] = "Расшифровать",
        ["common.contrasena"] = "ПАРОЛЬ",
        ["common.repetirContrasena"] = "ПОВТОРИТЕ ПАРОЛЬ",
        ["common.chk.borradoSeguro"] = "Безопасное удаление (перезаписать перед удалением)",
        ["common.chk.conservarOriginal.encriptar"] = "Сохранить исходный файл (не удалять по завершении)",
        ["common.chk.conservarOriginal.desencriptar"] = "Сохранить зашифрованный файл .enc (не удалять по завершении)",
        ["common.estado.passwordRequerida"] = "Введите пароль.",
        ["common.estado.passwordsNoCoinciden"] = "Пароли не совпадают.",
        ["common.estado.procesando"] = "Обработка...",
        ["common.estado.progreso"] = "{0}%",
        ["common.estado.progresoConEta"] = "{0}% · осталось {1}",
        ["common.estado.archivoDesencriptado"] = "Файл успешно расшифрован.",
        ["common.estado.archivoEncriptado"] = "Файл успешно зашифрован.",
        ["common.estado.carpetaEncriptada"] = "Папка зашифрована в один файл (включено файлов: {0}).",
        ["common.estado.variosEncriptados"] = "Файлов зашифровано в один пакет: {0}.",
        ["common.estado.carpetaRestaurada"] = "Папка восстановлена: {0} из {1} файлов.",
        ["common.estado.operacionCancelada"] = "Операция отменена. Ничего не изменено.",
        ["common.estado.verificandoPassword"] = "Проверка пароля...",
        ["common.estado.noPudoDesencriptar"] = "Не удалось расшифровать: неверный пароль или повреждённый файл.",
        ["common.estado.error"] = "Ошибка: {0}",
        ["common.footer.credito"] = "Разработано: {0}",
        ["common.contextmenu.encriptarCon"] = "Зашифровать с помощью Encriptador",
        ["common.contextmenu.progIdNombre"] = "Зашифрованный файл (Encriptador)",

        // ===== strength =====
        ["strength.muyDebil"] = "Очень слабый",
        ["strength.debil"] = "Слабый",
        ["strength.media"] = "Средний",
        ["strength.fuerte"] = "Сильный",
        ["strength.muyFuerte"] = "Очень сильный",

        // ===== crypto =====
        ["crypto.archivoDanado"] = "Неверный пароль или повреждённый файл.",
        ["crypto.rutaNoExiste"] = "Путь не существует.",
        ["crypto.tipoEsCarpeta"] = "Этот .enc содержит папку: откройте его в проводнике вместо прямой расшифровки.",
        ["crypto.tipoEsArchivo"] = "Этот .enc — отдельный файл: расшифруйте его напрямую вместо просмотра.",
        ["crypto.archivoNoManifiesto"] = "Файл отсутствует в манифесте.",
        ["crypto.seleccionarArchivoEnc"] = "Для расшифровки выберите файл .enc, а не папку.",

        // ===== main (MainWindow) =====
        ["main.tagline"] = "Шифрование файлов AES-256-GCM",
        ["main.archivo.etiqueta"] = "ФАЙЛ",
        ["main.btn.elegirCarpeta"] = "Папка",
        ["main.archivo.placeholder"] = "Перетащите файл(ы) или папку сюда, или нажмите, чтобы выбрать",
        ["main.archivo.carpeta"] = "Папка: {0}",
        ["main.archivo.variosSeleccionados"] = "Выбрано файлов: {0}",
        ["main.dialog.abrir.titulo"] = "Выбрать файл(ы)",
        ["main.dialog.abrir.filtroEtiqueta"] = "Все файлы (*.*)",
        ["main.dialog.guardar.titulo"] = "Сохранить как",
        ["main.dialog.guardar.filtroEtiqueta"] = "Зашифрованный файл (*.enc)",
        ["main.dialog.guardar.nombreVarios"] = "{0} файлов.enc",
        ["main.error.carpetaUnaPorVez"] = "Для папок выбирайте по одной за раз (нельзя сочетать с другими файлами).",
        ["main.error.soloUnArchivoEnc"] = "Для расшифровки выберите один файл .enc.",
        ["main.error.noEsCarpeta"] = "Для расшифровки выберите файл .enc (не папку).",
        ["main.error.seleccionInvalida"] = "Выберите допустимый файл или папку.",
        ["main.tooltip.acerca"] = "О программе",

        // ===== quick (QuickWindow) =====
        ["quick.archivo.archivo"] = "Файл: {0}",

        // ===== explorer (ExplorerWindow) =====
        ["explorer.titulo"] = "Зашифрованное содержимое",
        ["explorer.instrucciones"] = "Дважды щёлкните файл, чтобы открыть его. Ничего не сохраняется, пока вы не извлечёте выбранные файлы.",
        ["explorer.btn.seleccionarTodo"] = "Выбрать всё",
        ["explorer.btn.deseleccionarTodo"] = "Снять выделение",
        ["explorer.btn.cancelar"] = "Отмена",
        ["explorer.btn.extraerTodo"] = "Извлечь всё ({0})",
        ["explorer.btn.extraerSeleccionados"] = "Извлечь выбранные ({0})",
        ["explorer.subtitulo_uno"] = "1 файл",
        ["explorer.subtitulo_otros"] = "Файлов: {0}",
        ["explorer.tooltip.dobleClic"] = "Двойной клик, чтобы открыть",
        ["explorer.estado.abriendo"] = "   (открывается...)",
        ["explorer.estado.errorAlAbrir"] = "   (не удалось открыть: {0})",
        ["explorer.error.noSePudoAbrir"] = "Не удалось открыть файл: {0}",

        // ===== about (AboutWindow) =====
        ["about.desarrollador"] = "Разработчик приложения",
        ["about.tagline"] = "Шифрование и расшифровка файлов с помощью AES-256-GCM",
        ["about.version"] = "Версия 1.0 · Avalonia UI",
        ["about.cerrar"] = "Закрыть",
    };
}
